cephbackupstorage/cephbackupstorage/cephagent.py

```
@@ -1072,6 +1109,98 @@ def cancel(self, req):
+        rsp.directUploadPath = self.get_direct_upload_path(req[http.REQUEST_HEADER]['Host'])
+        return jsonobject.dumps(rsp)
+
+    @replyerror
+    def direct_upload_file(self, req):
+        try:
+            UploadHandler(req, self.upload_tasks).handle_upload()
+        except Exception as e:
+            logger.exception("File upload failed: %s", str(e))
+
+    @replyerror
+    def get_upload_progress(self, req):
```

Ruff (0.14.10)

Redefinition of unused `get_upload_progress` from line 681

(F811)